### PR TITLE
071421Graig Drop and Create Tables completed with one seeded album

### PR DESCRIPTION
**Pull request by rgmantle**
Created the tables for the DB
Note: I tried to make an Inventory table, but matching the product to the table was getting hard. I think maybe we can do an inventory table that just brings over the albumId, price from the products table and then the inventory and onhand amounts are automated with functions that change on sales like the click counter

### CODE REVIEW

File: db/init_db.js [hunk 1]
```diff
@@ -1,25 +1,74 @@
...
 
     // build tables in correct order
 
+    await client.query(`
+    DROP TABLE IF EXISTS users;
+    DROP TABLE IF EXISTS genre_albums;
+    DROP TABLE IF EXISTS genres;
+    DROP TABLE IF EXISTS products;
+
+    CREATE TABLE products (
```
Comment (banchito) [inline db/init_db.js:24]: I would prefer albums instead of products. or change genres_albums for genres_products.

File: db/index.js [hunk 2]
```diff
@@ -1,13 +1,87 @@
...
+  username,
+  password,
+  email
+}) {
+  try {
+    const { rows: [ user ] } = await client.query(`
+      INSERT INTO users(username, password, email)
+      VALUES($1, $2, $3)
+      ON CONFLICT (username) DO NOTHING
+      RETURNING *;
```
Comment (banchito) [inline db/index.js:70]: RETURNING id, username, email ?

File: db/init_db.js [hunk 3]
```diff
@@ -1,25 +1,74 @@
...
-
-    // drop tables in correct order
+    console.log("Building tables...");
+    await client.connect();
 
     // build tables in correct order
 
+    await client.query(`
+    DROP TABLE IF EXISTS users;
+    DROP TABLE IF EXISTS genre_albums;
```
Comment (banchito) [inline db/init_db.js:20]: genre_albums references products and genres so those tables need to be dropped first I think.

**Review by banchito — CHANGES_REQUESTED** [review]

File: db/index.js (hunk 2 above)
Comment (andreavanginneken) [reply db/index.js:70]: Maybe have reorder as a stretch goal? no worries either way tho

File: db/init_db.js [hunk 4]
```diff
@@ -1,25 +1,74 @@
...
+    DROP TABLE IF EXISTS genres;
+    DROP TABLE IF EXISTS products;
+
+    CREATE TABLE products (
+      id SERIAL PRIMARY KEY,
+      album_name varchar(255) UNIQUE NOT NULL,
```
Comment (burk96) [inline db/init_db.js:26]: UNIQUE might cause issues if two artists have albums of the same name (ex: "Greatest Hits").

File: db/index.js [hunk 5]
```diff
@@ -1,13 +1,87 @@
...
+  } catch (error) {
+    throw error;
+  }
+}
+
+async function createGenres({
+  genre,
+}) {
+  try {
+    const { rows: [ genrename ] } = await client.query(`
```
Comment (burk96) [inline db/index.js:36]: Nitpick I know but maybe camel case genreName?

File: db/init_db.js [hunk 6]
```diff
@@ -1,25 +1,74 @@
...
+      artist varchar(255) NOT NULL,
+      year INT,
+      price INT DEFAULT 1999,
+      quantity INT DEFAULT 0,
+      reorder_number INT DEFAULT 0,
+      img_url varchar(255) DEFAULT 'https://cdn4.vectorstock.com/i/thumb-large/82/48/vinyl-record-blank-realistic-vinyl-disc-mockup-on-vector-17128248.jpg'
```
Comment (burk96) [inline db/init_db.js:32]: Just saying I appreciate your use of DEFAULTs, I didn't realize that was a thing

File: db/init_db.js [hunk 7]
```diff
@@ -1,25 +1,74 @@
...
 
 async function populateInitialData() {
   try {
-    // create useful starting data
+    console.log("Starting to create albums...");
+    await createAlbums({
+      album_name: 'The Colour and the Shape',
+      artist: 'Foo Fighters',
+      year: 1997,
+      price: 1999,
```
Comment (burk96) [inline db/init_db.js:64]: Another nitpick, did you mean 19.99? Lol

File: db/init_db.js (hunk 6 above)
Comment (burk96) [reply db/init_db.js:32]: We might want to make img_url actually an array of image_urls. Currently in our JSON we have an array that looks like this: 
`"images": [
      {
        "height": 640,
        "url": "https://i.scdn.co/image/ab67616d0000b27376713b7514fdc2770ec12ab7",
        "width": 640
      },
      {
        "height": 300,
        "url": "https://i.scdn.co/image/ab67616d00001e0276713b7514fdc2770ec12ab7",
        "width": 300
      },
      {
        "height": 64,
        "url": "https://i.scdn.co/image/ab67616d0000485176713b7514fdc2770ec12ab7",
        "width": 64
      }
    ],`


If you think we should just store one image_url, ignore this and I'll change our Spotify JSON parser

**Review by burk96 — CHANGES_REQUESTED** [review]
I think you did a great job initializing our DB, I just have two minor nitpicks, one suggestion, one concern (UNIQUE), and one compliment 😃

File: db/index.js (hunk 5 above)
Comment (banchito) [reply db/index.js:36]: I refrained myself on this lolo

Comment (rgmantle) [reply db/index.js:36]: I guess. I was using username which is all lower case but ill change it

**Pull request merged**